### DISCUSSION TITLE
Apple Macbook M1 Air - invalid syntax - privateGPT.py - line 26

### DISCUSSION THREAD

**Discussion by davidtkeane** [imartinez/privateGPT#296] [Q&A] [answered]
I am getting this error when trying to run the privateGPT.py file when inside visual studio and also when i run the file from the command line using - python3 privateGPT.py

python3 privateGPT.py
  File "privateGPT.py", line 26
    match model_type:
          ^
SyntaxError: invalid syntax

[Running] python -u "/privateGPT/privateGPT.py"
  File "/privateGPT/privateGPT.py", line 26
    match model_type:
          ^
SyntaxError: invalid syntax

I am presumming I have all the other steps for install done, Did anyone else have this error at all? 

I am using a Mac Air M1, with visual studio. requirements.txt installed, .env file, models folder with the model inside, I ran ingest.py and it made a db folder within the privateGPT folder. I have tried python and python3 and same error. I checked github and the privateGPT.py file is the same as mine.  

I tried also to change the line match models to this and match model_type in ['LlamaCpp', 'GPT4All']: and also I used case model_type in ['LlamaCpp', 'GPT4All']: Both didn't work. Reverted back to origional code. 

If I find a work around then I will post back here with the find. 

Any assistance would be great! Thanks for the time you spent working this all out!

**Comment by johnbrisbin** [comment, marked as answer]
Your python version is too old.
"To use this software, you must have Python 3.10 or later installed. Earlier versions of Python will not compile."
Yours in 3.9 or earlier.
